posthog/api/person.py

```diff
@@ -397,33 +395,42 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> response.
         ],
     )
     def destroy(self, request: request.Request, pk=None, **kwargs):  # type: ignore
+        """
+        The deletion request will be completed via sending a "$delete_person" event.
```

Comment:
Wondering if we still want to do some sort of soft deletion here:

* Person deletion is now async, not guaranteed to complete in any particular timeframe.
* There could be ongoing requests to mutate a person with a $delete_person event onflight.
* All these requests will be wasting their time, as the person will eventually be deleted, and any work done will be lost.
* By marking the person as to be deleted after the event is sent we can:
  * Block new requests to mutate that person, saving resources.
  * Give immediate feedback to users, as the person can be skipped even if the `$delete_person` event is inflight (more critical in scenarios with pipeline lag).


Any thoughts?